src/test/java/com/framework/learn/automation/firsttests/AllTestNgAnnotationsTest.java

```diff
@@ -0,0 +1,132 @@
+ * @BeforeMethod //done in test
+ * The annotated method will be run before each test method.
+ * @AfterMethod // done in test
+ * The annotated method will be run after each test method.
+ * @DataProvider // done in test
+ * Marks a method as supplying data for a test method. The annotated method must return an Object[ ][ ], where each Object[ ] can be assigned the parameter list of the test method. The @Test method that wants to receive data from this DataProvider needs to use a dataProvider name equals to the name of this annotation.
+ * @Factory //done in factory
+ * Marks a method as a factory that returns objects that will be used by TestNG as Test classes. The method must return Object[ ].
+ * @Parameters /done in Suite.xml
+ * Describes how to pass parameters to a @Test method.
+ */
+
+public class AllTestNgAnnotationsTest extends BaseTest {
+
+    private String newMessage = "";
+
+    public AllTestNgAnnotationsTest(String newMessage) {
+        this.newMessage = newMessage;
+    }
+
+
+    @BeforeSuite(description = "Clean up")
+    public void cleanCookie() {
+        LOGGER.info(newMessage);
+        LOGGER.info(" ______@BeforeSuite______");
+    }
+
+    @BeforeClass(description = "set ub driver")
+    public void setUp() {
+        LOGGER.info(newMessage);
+        LOGGER.info(" ______@BeforeClass______");
+    }
+
+    @AfterClass(description = "After class")
+    public void tearDown() {
+        LOGGER.info(newMessage);
+        LOGGER.info(" ______@AfterClass______");
+    }
+
+    @AfterSuite(description = "Clean up", enabled = true)
+    public void cleanCookieAfter() {
+        LOGGER.info(newMessage);
+        LOGGER.info(" ______@AfterSuite______");
+    }
+
+    @BeforeTest(description = "Open url")
+    public void openUrl() {
+        LOGGER.info(newMessage);
+        LOGGER.info(" ______@BeforeTest______");
+    }
+
+    @BeforeGroups(description = "group of tests", groups = "group")
+    public void beforeGroupOfTests() {
+        LOGGER.info(newMessage);
+        LOGGER.info(" ______@BeforeGroup______");
+    }
+
+    @Test(description = "First Test", priority = 1, groups = "group", dataProvider = "Buttons")
+    public void testForAllAnnotations(String element) {
+        LOGGER.info(newMessage);
+        LOGGER.info(element);
+    }
+
+    @Parameters({"parameterFromXml"})
+    @Test(description = "Second Test", priority = 2)
+    public void testForAllAnnotationsWithParameters(String parameterFromXml) {
+        LOGGER.info(newMessage);
+        LOGGER.info(parameterFromXml);
+    }
+
+    @AfterGroups(description = "after group of tests", groups = "group")
+    public void beforeGroupRun() {
+        LOGGER.info(newMessage);
+        LOGGER.info(" ______@AfterGroup______");
+    }
+
+    @DataProvider(name = "Buttons")
+    public Iterator<Object[]> buttonProvider() {
```

Comment:
what can be return type here? in data provider 
[][] try with matrix 
to return two value for tests, not one